cli/src/pcluster/validators/secret_validators.py

```diff
@@ -0,0 +1,76 @@
+    """Validate that Arn is a valid ARN in given region."""
+
+    def _validate(self, arn: str, region: str, expected_service: str, expected_resource: str):
+        service, resource = _get_service_and_resource(arn)
+        if not (service == expected_service and resource == expected_resource):
+            self._add_failure(f"The {arn} is not supported in region {region}.", FailureLevel.ERROR)
+
+
+class MungeKeySecretSizeAndBase64Validator(Validator):
+    """Validate that MungeKeySecretArn exists.
+
+    In Base64 encoding:
+    - Every 3 bytes of input are represented as 4 characters in the encoded string.
+    - The length of the encoded string must be a multiple of 4.
+    - Valid Base64 characters include upper/lowercase letters,
+      numbers, '+', '/', and possibly trailing '=' padding chars.
+    - The '=' padding is used if the number of bytes being encoded is not divisible by 3.
+    Given these rules:
+    - "validBase64" is valid because its length is a multiple of 4 and uses valid Base64 characters.
+    - "invalidBase64" is invalid because its length is not a multiple of 4.
+
+    Also:
+    - The size of the decoded munge key must between 256 and 8192 bits.
+    """
+
+    def _validate(self, munge_key_secret_arn: str):
+        try:
+            # Get the actual key value to check if it is a valid Base64
+            secret_response = AWSApi.instance().secretsmanager.get_secret_value(munge_key_secret_arn)
+            secret_value = secret_response.get("SecretString")
+
+            if not secret_value:
+                self._add_failure(
+                    f"The secret {munge_key_secret_arn} does not contain a valid secret string.",
+                    FailureLevel.ERROR,
+                )
+                return
+
+            try:
+                # Try decoding key value from Base64
+                decoded_secret = base64.b64decode(secret_value, validate=True)
+                # Check if the size of the decoding key is within the accepted range
+                decoded_secret_size_in_bits = len(decoded_secret) * 8
+                if decoded_secret_size_in_bits < 256 or decoded_secret_size_in_bits > 8192:
+                    self._add_failure(
+                        f"The size of the decoded munge key in the secret {munge_key_secret_arn} is "
+                        f"{decoded_secret_size_in_bits} bits. Please use a key with a size between 256 and 8192 bits.",
+                        FailureLevel.ERROR,
+                    )
+            except binascii.Error:
+                self._add_failure(
+                    f"The content of the secret {munge_key_secret_arn} is not a valid Base64 encoded string. "
+                    f"Please refer to the ParallelCluster official documentation for more information.",
+                    FailureLevel.ERROR,
+                )
+        except AWSClientError as e:
+            _handle_arn_aws_client_error(e, munge_key_secret_arn, self)
```

Comment:
Are you sure this function (which was derived from the directory service `PasswordSecretArnValidator` can cope with the failures from the `get_secret_value()` API call?

Reply:
I didn't change the logic inside it compared to before. Also didn't change the unit test for this Validator, and the unit test can pass. But anyway, I will do a manually test for it tomorrow.

Reply:
Did you perform a test in then end?